esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,172 @@
+            raise DbConnectionError(self.db_path)
+
+    def create_tables(self):
+        """
+        Create the 3 tables used for per-datastore authorization.  This function
+        should only be called once per datastore.  It will raise an exception if
+        the schema file isn't accessible or the tables already exist.
+        """
+        with open('./docker-volume-auth-schema.sql') as f:
+            sql = f.read()
+            self.conn.executescript(sql)
+            self.conn.commit()
+
+    def create_tenant(self, name, description, vms, privileges):
+        """
+        Create a tenant in the database. A tenant id will be auto-generated and
+        returned. vms are (vm_id, vm_name) pairs. Privileges are dictionaries
+        with keys matching the row names in the privileges table. Tenant id is
+        filled in for both the vm and privileges tables.
+        """
+
+        # Create the entry in the tenants table
+        id = str(uuid.uuid4())
+        self.conn.execute(
+            "insert into tenants(id, name, description) values (?, ?, ?)",
+            (id, name, description)
+        )
+
+        # Create the entries in the vms table
+        vms = [(vm_id, vm_name, id) for (vm_id, vm_name) in vms]
+        print vms
+        if vms:
+            self.conn.executemany(
+              "insert into vms(vm_id, vm_name, tenant_id) values (?, ?, ?)",
+              vms
+            )
+
+        # Create the entries in the privileges table
+        # TODO: data scrubbing, ensure all keys exist
+        for p in privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+        
+        self.conn.commit()
+        return id
+
+    def add_vm_to_tenant(self, vm_id, vm_name, tenant_id):
+            self.conn.execute(
+                "insert into vms(vm_id, vm_name, tenant_id) values (?, ?, ?)",
+                (vm_id, vm_name, tenant_id)
+            )
+            self.conn.commit()
+
+    def remove_vm_from_tenant(self, vm_id, vm_name, tenant_id):
+            self.conn.execute(
+                "delete from vms where vm_id=? AND tenant_id=?", 
+                (vm_id, tenant_id)
+            )
+            self.conn.commit()
+
+    def set_name(self, name, tenant_id):
+        self.conn.execute(
+                "update tenants set name=? where id=?", 
+                (name, tenant_id)
+            )
+        self.conn.commit()
+    
+    def set_description(self, description, tenant_id):
+        self.conn.execute(
+                "update tenants set description=? where id=?", 
+                (description, tenant_id)
+            )
+        self.conn.commit()
+
+    def set_datastore_access_privileges(self, tenant_id, privileges):
```

Comment:
`datastore` shouldn't be it's own parameter. It isn't part of the privileges. It's part of the key `(tenant_id, datastore)`. This also clears up the comment below. The call should be made twice: once for each tenant, datastore pair.